File: java/src/test/java/com/alxarafe/domain/rules/valueobject/RuleScopeTest.java
package com.alxarafe.domain.rules.valueobject;

import com.alxarafe.app.domain.rules.valueobject.RuleScope;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RuleScopeTest {

    @Test
    void values() {
        assertEquals(RuleScope.LOCATION, RuleScope.valueOf("LOCATION"));
        assertEquals(RuleScope.AISLE, RuleScope.valueOf("AISLE"));
        assertEquals(RuleScope.ZONE, RuleScope.valueOf("ZONE"));
        assertEquals(3, RuleScope.values().length);
    }
}

File: php/tests/Unit/Domain/Rules/ValueObject/RuleScopeTest.php
<?php
declare(strict_types=1);

namespace Tests\Unit\Domain\Rules\ValueObject;

use Alxarafe\App\Domain\Rules\ValueObject\RuleScope;
use PHPUnit\Framework\TestCase;

final class RuleScopeTest extends TestCase
{
    public function testValues(): void
    {
        self::assertSame('LOCATION', RuleScope::LOCATION->value);
        self::assertSame('AISLE', RuleScope::AISLE->value);
        self::assertSame('ZONE', RuleScope::ZONE->value);
    }
}
